File: frontend/src/services/openapi/models/UpcomingCompetitorModel.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

import type { CompetitorModel } from './CompetitorModel';

/**
 * Represents an upcoming competitor.
 */
export type UpcomingCompetitorModel = {
    /**
     * Unique ID for these competitors
     */
    id: number;
    /**
     * Competitor(s). Contains at least one entity.
     */
    competitors: Array<CompetitorModel>;
}

File: src/Api.Models/UpcomingCompetitorModel.cs
using System.ComponentModel.DataAnnotations;

namespace Api.Models
{
    /// <summary>
    /// Represents an upcoming competitor.
    /// </summary>
    public class UpcomingCompetitorModel
    {
        /// <summary>
        /// Unique ID for these competitors
        /// </summary>
        /// <example>123</example>
        [Required]
        public int Id { get; set; }

        /// <summary>
        /// Competitor(s). Contains at least one entity.
        /// </summary>
        [Required]
        public CompetitorModel[] Competitors { get; set; } = null!;
    }
}
